contracts/Aavegotchi/facets/ItemsFacet.sol

```diff
@@ -298,58 +289,59 @@ contract ItemsFacet is Modifiers {
+            _gotchiInfo.equippedDelegatedItemsCount += 1;
             s.depositIdToEquippedGotchis[_depositId.grantor][_depositId.nonce].add(_gotchiId);
         } else {
-            require(s.ownerItemBalances[_sender][_toEquipWearableId] >= _balToTransfer, "ItemsFacet: Wearable isn't in inventory");
+            require(s.ownerItemBalances[_sender][_toEquipWearableId] >= 1, "ItemsFacet: Wearable isn't in inventory");
 
-            LibItems.removeFromOwner(_sender, _toEquipWearableId, _balToTransfer);
-            IEventHandlerFacet(s.wearableDiamond).emitTransferSingleEvent(_sender, _sender, address(this), _toEquipWearableId, _balToTransfer);
+            LibItems.removeFromOwner(_sender, _toEquipWearableId, 1);
+            IEventHandlerFacet(s.wearableDiamond).emitTransferSingleEvent(_sender, _sender, address(this), _toEquipWearableId, 1);
             LibERC1155Marketplace.updateERC1155Listing(address(this), _toEquipWearableId, _sender);
         }
-
-        LibItems.addToParent(address(this), _gotchiId, _toEquipWearableId, _balToTransfer);
-        emit LibERC1155.TransferToParent(address(this), _gotchiId, _toEquipWearableId, _balToTransfer);
+       
+        LibItems.addToParent(address(this), _gotchiId, _toEquipWearableId, 1);
+        emit LibERC1155.TransferToParent(address(this), _gotchiId, _toEquipWearableId, 1);
     }
     
     function _removeWearableFromGotchi(
         uint256 _gotchiId,
-        uint256 _existingEquippedWearableId
+        uint256 _existingEquippedWearableId,
+        uint256 _slot,
+        GotchiEquippedItemsInfo storage _gotchiInfo
     ) internal {
-        GotchiEquippedItemsInfo storage _equippedItemsInfo = s.gotchiEquippedItemsInfo[_gotchiId];
-        address _sender = LibMeta.msgSender();
-
+       
         LibItems.removeFromParent(address(this), _gotchiId, _existingEquippedWearableId, 1);
         emit LibERC1155.TransferFromParent(address(this), _gotchiId, _existingEquippedWearableId, 1);
-
-        ItemDepositId storage _depositId = _equippedItemsInfo.equippedItemIdToDelegationInfo[_existingEquippedWearableId].depositId;
+        
+        address _sender = LibMeta.msgSender();
+        ItemDepositId storage _depositId = _gotchiInfo.equippedDelegatedItems[_slot];
+        
         if (_depositId.nonce != 0) {
             // remove wearable from Aavegotchi and delete delegation
-            uint256 _delegatedBalance = _equippedItemsInfo.equippedItemIdToDelegationInfo[_existingEquippedWearableId].balance;
 
-            if(_delegatedBalance == 1) {
-                delete _equippedItemsInfo.equippedItemIdToDelegationInfo[_existingEquippedWearableId];
+            if(!(_slot == LibItems.WEARABLE_SLOT_HAND_LEFT 
+                && (_gotchiInfo.equippedDelegatedItems[LibItems.WEARABLE_SLOT_HAND_RIGHT].nonce == _depositId.nonce
+                && _gotchiInfo.equippedDelegatedItems[LibItems.WEARABLE_SLOT_HAND_RIGHT].grantor == _depositId.grantor))) {
```

Comment:
what if we are only unequipping the `WEARABLE_SLOT_HAND_RIGHT`? we need to review this `if`